server/src/main/java/org/elasticsearch/indices/PostRecoveryMerger.java

```diff
@@ -44,14 +46,14 @@ class PostRecoveryMerger {
     private static final boolean TRIGGER_MERGE_AFTER_RECOVERY;
 
     static {
-        final var propertyValue = System.getProperty("es.trigger_merge_after_recovery");
+        final var propertyValue = System.getProperty("es.trigger_merge_after_recovery_8_515_00_0");
```

Comment:
Renaming the kill-switch here to nullify the overrides that are in place today and automatically re-enable this feature when upgrading. We can rename it back again later.